accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,320 @@
+
+# Installation operations
+
+## Workload pack tracking and installation cleanup
+
+Workload artifacts should not be left over after uninstalling or updating workloads.  This is complicated by the fact that the same workload pack may be used by different workloads, and even different .NET SDK feature bands.  It will be the responsibility of the installer abstraction implementation to manage this.  
+
+For installer abstraction implementations that wrap installers where the workload will appear in a user-visible fashion (for example as a checkbox in the Visual Studio installer, an entry in Windows Add/Remove programs, or as a package installed on a Linux OS), the installer abstraction implementation will generally rely on the underlying installer technology to manage the cleanup of artifacts which may be shared with other workload installations.
+
+Installer abstraction implementations that install workload packs individually will also need to remove those packs when they are no longer needed.  After a workload installation operation (which includes updating or uninstalling workloads), the installer abstraction implementation should remove workload packs which are no longer needed.  To support this, they will generally need to keep a record of which workload packs have been installed for which SDK feature band, along with whether the pack is still in use for that band.  This information should be kept in a form that is likely to be forward and backwards compatible across different SDK versions.  First, the workload pack installation records for any SDK feature bands which are no longer installed can be removed.  Then any workload pack installation records which are not currently used by an installed workload for the current band can be marked as not in use.  Then, any workload packs that don't have any installation records marked as in use can be uninstalled.
+
+## SDK update process
+When the .NET SDK is updated to a new version, developers may expect for the same workloads to be installed for the new version as were installed for the old version.  We probably won't be able to do this for all installer technologies.  Nevertheless, the ideal behavior would be this:
+
+- For in-place updates to the .NET SDK, we should also update the workloads when we update the SDK
+  - For SDK installations managed by Visual Studio, in-place updates can include updating to a new SDK feature band.  In that case the workload installation is managed by Visual Studio, so the workloads will be updated as we want with the plans we have today
+- When installing a new version of the .NET SDK side-by-side with an existing version, we should default to installing the same workloads that were installed for the previous version.  The developer should be able to override this, for example by unchecking the corresponding checkboxes in an installer UI
```

Comment:
Is this a new major SDK release, or a new feature band release? For example, if this user is running `5.1.1xx`, is this about going to `5.1.2xx` or `6.0.1xx`? Or both?

Reply:
This would be for both.